Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -960,6 +995,56 @@ static String mutationObserverNotificationString()
     }
 }
 
+#if ENABLE(UNIFIED_PDF_PAINT_SELECTIONS)
+void UnifiedPDFPlugin::paintPDFSelection(GraphicsContext& context, const FloatRect& clipRect)
+{
+    if (!m_currentSelection || [m_currentSelection isEmpty])
+        return;
+
+    bool isVisibleAndActive = false;
+    if (RefPtr page = this->page())
+        isVisibleAndActive = page->isVisibleAndActive();
+
+    auto& rendererTheme = m_element->renderer()->theme();
+    auto styleColorOptions = m_element->renderer()->styleColorOptions();
+    auto selectionColor = isVisibleAndActive ? rendererTheme.activeSelectionBackgroundColor(styleColorOptions) : rendererTheme.inactiveSelectionBackgroundColor(styleColorOptions);
+
+    auto pageCoverage = pageCoverageForRect(clipRect);
+    auto documentScale = pageCoverage.pdfDocumentScale;
+    for (auto& pageInfo : pageCoverage.pages) {
+        auto page = m_documentLayout.pageAtIndex(pageInfo.pageIndex);
+        if (!page || !shouldDisplayPage(pageInfo.pageIndex))
+            continue;
+
+        auto pageDestinationRect = pageInfo.pageBounds;
+
+        GraphicsContextStateSaver pageStateSaver { context };
+        context.scale(documentScale);
+        context.clip(pageDestinationRect);
+        // Translate the context to the bottom of pageBounds and flip, so that PDFKit operates
+        // from this page's drawing origin.
+        context.translate(pageDestinationRect.minXMaxYCorner());
+        context.scale({ 1, -1 });
+
+        auto pageGeometry = m_documentLayout.geometryForPage(page);
+        auto transformForBox = m_documentLayout.toPageTransform(*pageGeometry).inverse().value_or(AffineTransform { });
+        context.concatCTM(transformForBox);
+
+        if ([m_currentSelection respondsToSelector:@selector(enumerateRectsAndTransformsForPage:usingBlock:)]) {
+            [protectedCurrentSelection() enumerateRectsAndTransformsForPage:page.get() usingBlock:[&context, &selectionColor](CGRect cgRect, CGAffineTransform cgTransform) mutable {
+                // FIXME: Perf optimization -- consider coalescing rects by transform.
+                GraphicsContextStateSaver individualRectTransformPairStateSaver { context };
+
+                if (AffineTransform transform { cgTransform }; !transform.isIdentity())
+                    context.concatCTM(transform);
+
+                context.fillRect({ cgRect }, selectionColor, CompositeOperator::SourceOver, BlendMode::Multiply);
```

Comment:
This is wrong when painting into the selection layer; the layer itself needs to have the Multiply blend mode set, and you should paint opaquely. Overlapping selection rects should not show the overlap.

Reply:
Addressed in https://github.com/WebKit/WebKit/commit/3175c379f8966936bbf4d9833ad4b1c11e7a1ede